core/src/main/java/org/elasticsearch/index/query/PercolateQueryBuilder.java

```diff
@@ -459,4 +472,83 @@ public Weight createNormalizedWeight(Query query, boolean needsScores) throws IO
         }
     }
 
+    private static PercolateQuery.QueryStore createStore(PercolatorFieldMapper.PercolatorFieldType fieldType,
+                                                         QueryShardContext context,
+                                                         boolean mapUnmappedFieldsAsString) {
+        return ctx -> {
+            LeafReader leafReader = ctx.reader();
+            BinaryDocValues binaryDocValues = leafReader.getBinaryDocValues(fieldType.getQueryBuilderFieldName());
```

Comment:
Do we need to check for null? It could be null if no queries were indexed so far I believe?